src/coverlet.core/Coverage.cs

```diff
@@ -34,15 +34,16 @@ public Coverage(string module, string[] filters, string[] excludes)
         public void PrepareModules()
         {
             string[] modules = InstrumentationHelper.GetCoverableModules(_module);
-            string[] excludes =  InstrumentationHelper.GetExcludedFiles(_excludes);
+            //var ef = InstrumentationHelper.GetExcludedFiles(_excludes);
+            var excludedFilesHelper = new ExcludedFilesHelper(_excludes);
             _filters = _filters?.Where(f => InstrumentationHelper.IsValidFilterExpression(f)).ToArray();
 
             foreach (var module in modules)
             {
                 if (InstrumentationHelper.IsModuleExcluded(module, _filters))
                     continue;
 
-                var instrumenter = new Instrumenter(module, _identifier, _filters, excludes);
+                var instrumenter = new Instrumenter(module, _identifier, _filters, excludedFilesHelper);
```

Comment:
I would prefer that `_excludes` be passed in here as is and leave it to the `Instrumenter` class to instantiate `ExcludedFilesHelper`

Reply:
ok when you'll finish review i'll do all amendments, thank you for looking into this.